depocs/__init__.py

```diff
@@ -192,10 +192,11 @@ def open(self, call_site_level=1):
         self._Scoped__is_open = True
         self._Scoped__is_used = True
 
-        stack = inspect.stack()
-        if len(stack) > call_site_level:
-            self._Scoped__open_site = stack[call_site_level]
-
+        try:
+            self._Scoped__open_site = sys._getframe(call_site_level)
+        except ValueError:
+            pass
```

Comment:
It raises that error if call_site_level >= len(stack) so I left it a noop to match the previous logic.